src/lib.rs

```diff
@@ -190,313 +182,415 @@ pub struct AstarteDeviceDataEvent {
-    ///
-    ///     tokio::spawn(async move {
-    ///         while let Some(event) = rx_events.recv().await {
-    ///             assert!(event.is_ok());
-    ///         };
-    ///     });
-    ///
-    ///     device.handle_events().await;
-    /// }
-    /// ```
-    pub async fn handle_events(&mut self) -> Result<(), Error> {
-        loop {
-            // Don't keep a ref to lock so we can retry in the match bellow
-            let res = {
-                let mut lock = self.eventloop.lock().await;
-                lock.poll().await
-            };
-
-            let event = match res {
-                Ok(event) => event,
-                Err(err) => {
-                    error!("couldn't poll the event loop: {err:#?}");
-
-                    DelayedPoll::retry_poll_event(self).await?
-                }
-            };
-
-            // Keep consuming and processing packets until we have data for the user
-            match event {
-                Event::Incoming(incoming) => {
-                    trace!("MQTT Incoming = {:?}", incoming);
-
-                    match incoming {
-                        rumqttc::Packet::ConnAck(conn_ack) => {
-                            self.connack(conn_ack).await?;
-                        }
-                        rumqttc::Packet::Publish(publish) => {
-                            let mut sdk = self.clone();
-
-                            // New task for the publish to not block the event loop
-                            tokio::spawn(async move {
-                                let res = sdk.handle_publish(publish).await;
-
-                                let res = match res {
-                                    // Skip events like purge_properties
-                                    Ok(None) => return,
-                                    Ok(Some(res)) => Ok(res),
-                                    Err(err) => {
-                                        error!(
-                                            "error encountered while handling a publish: {err:#?}"
-                                        );
-
-                                        Err(err)
-                                    }
-                                };
-
-                                sdk.events_channel
-                                    .send(res)
-                                    .await
-                                    .expect("error channel dropped");
-                            });
-                        }
-                        packet => {
-                            trace!("packet received {packet:?}");
-                        }
-                    }
-                }
-                Event::Outgoing(o) => trace!("MQTT Outgoing = {:?}", o),
-            }
+impl<S, C> AstarteDeviceSdk<S, C> {
+    pub(crate) fn new(interfaces: Interfaces, store: S, connection: C, tx: EventSender) -> Self
+    where
+        S: PropertyStore,
```

Comment:
Are the trait bound needed?

Reply:
In the end we decided to add also the connection bound in the new function.